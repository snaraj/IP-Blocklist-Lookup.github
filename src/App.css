body {
	background: #22223B;
}

.particles {
	position: fixed;
	top: 0;
	right: 0;
	bottom: 0;
	left: 0;
	z-index: -1;
}

.select {
	display: flex;
	justify-content: center;
	align-items: center;
}