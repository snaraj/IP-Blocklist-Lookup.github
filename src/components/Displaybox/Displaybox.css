.text {
	font-family: 'Merriweather', serif;
	color: #F2E9E4;
	text-align: center;
	font-family: 'Merriweather', serif;
	text-shadow: 2px 2px 2px black;
	font-weight: bold;
}