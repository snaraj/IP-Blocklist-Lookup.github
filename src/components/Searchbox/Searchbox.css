.search-box {
	display: flex;
	justify-content: center;
	align-items: center;
	width: 100%;
	height: 100px;
}

.search-text {
	font-family: 'Merriweather', serif;
	font-style: italic;
	font-size: 4em;
	color: black;
	text-align: center;
}