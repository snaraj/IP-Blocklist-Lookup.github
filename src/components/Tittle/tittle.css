.tittle-box {
	display: flex;
	justify-content: center;
	align-items: center;
	width: 100%;
	height: 150px;
}

.tittle-text {
	font-family: 'Merriweather', serif;
/*	font-style: italic;
*/	font-size: 4em;
	color: #F2E9E4;
	text-shadow: 3px 3px 3px black;
	font-weight: bold;
}