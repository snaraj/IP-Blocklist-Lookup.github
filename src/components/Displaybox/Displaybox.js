import React from 'react';
import './Displaybox.css';

const Displaybox = (props) => {

	const getFilteredIps = ()  => {
		return props.ips.filter((ip) => {
			const stringIp = ip[0];
			return stringIp.includes(props.searchFilter);
		}).slice(0, props.limit);
	}

	return (
		<div className='text'>
			{JSON.stringify(getFilteredIps())}
		</div>
	)
}

export default Displaybox;