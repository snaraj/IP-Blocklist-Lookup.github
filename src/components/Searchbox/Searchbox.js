import React from 'react';
import './Searchbox.css';

const Searchbox = ({searchChange}) => {
	return (
		<div className='pa2 search-box'>
			<input 
				className='pa3 ba b--green bg-lightest-blue search-text'
				type='search' 
				placeholder='Filter'
				onChange={searchChange} 
			/>
		</div>
	)
}

export default Searchbox;