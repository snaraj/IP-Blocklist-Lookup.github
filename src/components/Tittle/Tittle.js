import React from 'react';
import './tittle.css';

const Tittle = () => {
	return (
		<div className='tittle-box'>
			<h1 className='tittle-text'>Raw IPs for the firewall block lists</h1>
		</div>
	)
}

export default Tittle;