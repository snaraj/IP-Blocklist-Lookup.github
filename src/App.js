import React, { Component } from 'react';
import './App.css';
import Tittle from './components/Tittle/Tittle';
import Searchbox from './components/Searchbox/Searchbox';
import Displaybox from './components/Displaybox/Displaybox';
import Scroll from './components/Scroll/Scroll';

class App extends Component {
	
	constructor() {
		super()
		this.state = {
			ips: [],
			searchfield: '',
			limit: 50,
		}
	}
	
	componentDidMount() {
		fetch('http://localhost:3000/').then(response => {
			return response.json();
		}).then(ip => {
			this.setState({ips: ip})
		})
	}

	onSearchChange = (event) => {
		this.setState({searchfield: event.target.value})
	}

	handleLimitChange = (e) => {
		this.setState({ limit: Number(e.target.value) });
	}

	render() {
    	if(this.state.ips.length === 0) {
      			return <h1>Loading...</h1>		
      	} else {
      		return (
	 			<div>
	  				<Tittle />
	  				<Searchbox searchChange={this.onSearchChange}/>
	  				<div className='select'>
	  					<select onChange={this.handleLimitChange}>
	  						<option value="50">50</option>
	  						<option value="100">100</option>
	  						<option value="200">200</option>
	  					</select>
	  				</div>
	  				<Scroll>
	  					<Displaybox limit={this.state.limit} searchFilter={this.state.searchfield} ips = {this.state.ips}/>
	  				</Scroll>
				</div>
    		);
      	}
    }
}

export default App;
