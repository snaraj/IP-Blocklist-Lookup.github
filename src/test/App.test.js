import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';
import App from '../App';
import Displaybox from '../components/Displaybox/Displaybox';
import Tittle from '../components/Tittle/Tittle';
import Searchbox from '../components/Searchbox/Searchbox';

configure({ adapter: new Adapter() });

describe('App', () => {
	it('App renders correctly in "debug" mode', () => {
		const component = shallow(<App debug />);
		expect(component).toMatchSnapshot();
	});
});

describe('Tittle', () => {
	it('Tittle renders correctly', () => {
		const component = shallow(<Tittle />);
		expect(component).toMatchSnapshot();
	});
});

describe('Displaybox', () => {
	it('Diplays something given a limit, a filter and a list of ips', () => {
		const component = shallow(<Displaybox limit={50} searchFilter={''} ips = {['131.204.192.84', '43.117.236.241']} />);
		expect(component.find('.text')).toBeDefined();
	});
});

describe('Searchbox', () => {
	it('handles a searchChange input', () => {
		const component = shallow(<Searchbox searchChange={'10'} />);
		expect(component.find('.search-box')).toBeDefined();
	})
})

//testing endpoint
describe('Testing endpoint', () => {
	it('Should get ip file', () => {
		return fetch('http://localhost:3000/').then(response => {
			return response.json();
		}).then(ip => {
			expect(ip).toBeDefined()
		});
	});
});