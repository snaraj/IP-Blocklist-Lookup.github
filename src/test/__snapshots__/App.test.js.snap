// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App App renders correctly in "debug" mode 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    debug={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Loading...",
    },
    "ref": null,
    "rendered": "Loading...",
    "type": "h1",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Loading...",
      },
      "ref": null,
      "rendered": "Loading...",
      "type": "h1",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Tittle Tittle renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Tittle />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <h1
        className="tittle-text"
      >
        Raw IPs for the firewall block lists
      </h1>,
      "className": "tittle-box",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Raw IPs for the firewall block lists",
        "className": "tittle-text",
      },
      "ref": null,
      "rendered": "Raw IPs for the firewall block lists",
      "type": "h1",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <h1
          className="tittle-text"
        >
          Raw IPs for the firewall block lists
        </h1>,
        "className": "tittle-box",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Raw IPs for the firewall block lists",
          "className": "tittle-text",
        },
        "ref": null,
        "rendered": "Raw IPs for the firewall block lists",
        "type": "h1",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Tittle Tittle renders correctly in "debug" mode 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Tittle
    debug={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <h1
        className="tittle-text"
      >
        Raw IPs for the firewall block lists
      </h1>,
      "className": "tittle-box",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Raw IPs for the firewall block lists",
        "className": "tittle-text",
      },
      "ref": null,
      "rendered": "Raw IPs for the firewall block lists",
      "type": "h1",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <h1
          className="tittle-text"
        >
          Raw IPs for the firewall block lists
        </h1>,
        "className": "tittle-box",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Raw IPs for the firewall block lists",
          "className": "tittle-text",
        },
        "ref": null,
        "rendered": "Raw IPs for the firewall block lists",
        "type": "h1",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
